value.go

```diff
@@ -194,9 +199,21 @@ func (v *MrbValue) Hash() *Hash {
 
 // String returns the "to_s" result of this value.
 func (v *MrbValue) String() string {
-	value := C.mrb_obj_as_string(v.state, v.value)
-	result := C.GoString(C.mrb_string_value_ptr(v.state, value))
-	return result
+	if v.IsDead() {
+		return ""
+	}
+
+	value, err := C.mrb_obj_as_string(v.state, v.value)
+	if err != nil {
+		return ""
```

Comment:
Here and below I'd return `<error retrieving string value: %s>` so we don't just lose the error.